### PR TITLE
rangeSlider component

### PR DESCRIPTION
**Pull request by zlatanpham**
This PR aims to create RangeSlider which will be used to narrow down search results, for example, based on pricing.

- [x] `rangeFilter`
- [x] `rangeAggregrateFilter`
- [x] `RangeSlider` component
- [x] ability to work with `rangeAggregrateFilter`
- [x] being customized in terms of styles, extra UI for tip, etc

### CODE REVIEW

**Comment by sampotts** [comment]
@zlatanpham I'll review once the styling is done perhaps? (cc @jschumacher)

**Comment by zlatanpham** [comment]
> I'll review once the styling is done perhaps?
Yes, there are a couple of small issues with the AggregrateFilter as well. I'll request your review later once getting those done.

File: src/components/RangeSlider/RangeSliderUI.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
+import { CSSObject } from "@emotion/core";
 import classnames from "classnames";
 import { Handle, RangeProps } from "rc-slider";
 import React from "react";
-import { CSSObject } from "styled-components";
```
Comment (zlatanpham) [inline src/components/RangeSlider/RangeSliderUI.tsx]: Ahh, we use `emotion` for styling but I used the autocomplete function from VSCode to inject the type and it wrongly picked the one from `styled-components`. I've pushed the code to fix that.

**Comment by zlatanpham** [comment]
> ## Handles can overtake each other
> I think we shouldn't allow the handles to overtake each other. So when the left handle can only butt up against the right handle rather than going past it.
> 
> ## Formatted numbers (nice to have)
> It'd be nice if the numbers were humanized with thousands seperators but this is a nice to have if it's a big bit of work. It's easily possible using [`Intl.NumberFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat) but if we have to support super old IE then it'd need polyfilling. I wasn't sure if we had something for it already?

@sampotts have you had the chance to look at the docs - https://deploy-preview-124--sajari-sdk-react.netlify.app/components/rangeSlider? We can custom the overtake behavior via `pushable` and `allowCross` prop and custom the format of the number by passing a custom `jsx` through `renderTip`.

**Comment by sampotts** [comment]
> @sampotts have you had the chance to look at the docs - https://deploy-preview-124--sajari-sdk-react.netlify.app/components/rangeSlider? We can custom the overtake behavior via `pushable` and `allowCross` prop and custom the format of the number by passing a custom `jsx` through `renderTip`.

Ah, I didn't know we had examples in the docs. That's fine then 👍

**Comment by zlatanpham** [comment]
> # Proposed design
> react-sdk-slider
> 
> This design takes into account the above issues. The left handle is the `:focus`/`:active` state. The fill color should use their theme color and we can default it to whatever we use for the other components. If there's no color then grab me on Slack and we can work something out. There is 40px padding/margin above the track and 15px below. It doesn't have to match exactly but close would be good to give the slider some breathing space.

- Yes, you're right about the spacing.
- The current implement does handle `:focus` state but yes, it doesn't work properly like you said. One thing I just notice the focus/active state will disappear once you use left/right arrow to control the handle. Is that the thing you want me to fix?

Anyway, will update the style to follow the new design 😃

**Comment by sampotts** [comment]
>     * The current implement does handle `:focus` state but yes, it doesn't work properly like you said. One thing I just notice the focus/active state will disappear once you use left/right arrow to control the handle. Is that the thing you want me to fix?

Yes please, that'd be good if it kept the focus style while using keyboard input.

**Comment by sampotts** [comment]
Looks good to me 👍 Nice one. I guess we just need @benhinchley's seal of approval and we're good?

File: src/components/RangeSlider/rangeSlider.mdx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+---
+
+import { PropsTable } from "docz";
+import { RangeFilter } from "../../controllers";
+import { RangeSlider } from "./RangeSlider.tsx";
+import { Playground } from "docz";
+import { EVENT_SELECTION_UPDATED } from "../../events";
+
+# RangeSlider
+
```
Comment (benhinchley) [inline src/components/RangeSlider/rangeSlider.mdx:14]: The `RangeSlider` component provides a slider which constructs the filter for the search request. It is customizable through the use of props which allows you to apply different styles, set different step value, enable or disable crossover, restrict a margin, and render a prefix.

Comment (benhinchley) [reply src/components/RangeSlider/rangeSlider.mdx:14]: ^ Sorry @zlatanpham somehow lost the message I wrote on the review comment, if you could just add the above paragraph in this change is good to go.

Comment (zlatanpham) [reply src/components/RangeSlider/rangeSlider.mdx:14]: I've added the message.

**Comment by sampotts** [comment]
>     * I needed to add `theme: any;` in the `ContainerProps` interface in `src/components/Tabs/Tab/styled.ts` for the build to work.

Hey @zlatanpham, did this get sorted? ☝️

**Comment by zlatanpham** [comment]
> > ```
> > * I needed to add `theme: any;` in the `ContainerProps` interface in `src/components/Tabs/Tab/styled.ts` for the build to work.
> > ```
> 
> 
> Hey @zlatanpham, did this get sorted? ☝️

@sampotts I mentioned the reason here - https://github.com/sajari/sajari-sdk-react/pull/124#discussion_r411090692 plus the commit to fix it - https://github.com/sajari/sajari-sdk-react/pull/124/commits/f895cc7ce9b1b02cfa1fdd8284f601c50d342e99. Are you still experiencing the issue?

**Comment by sampotts** [comment]
> @sampotts I mentioned the reason here - [#124 (comment)](https://github.com/sajari/sajari-sdk-react/pull/124#discussion_r411090692) plus the commit to fix it - [f895cc7](https://github.com/sajari/sajari-sdk-react/commit/f895cc7ce9b1b02cfa1fdd8284f601c50d342e99). Are you still experiencing the issue?

Yeah I'm still seeing it on latest. A customer also raised it. I've opened a PR here with a fix and also an unrelated docs update: 
https://github.com/sajari/sajari-sdk-react/pull/125

**Pull request merged**